tools/functional-tester/etcd-tester/checks.go

```diff
@@ -65,6 +66,129 @@ func (hc *hashChecker) Check() (err error) {
 	return nil
 }
 
+type leaseChecker struct {
+	leaseStressers []Stresser
+}
+
+func newLeaseChecker(leaseStressers []Stresser) Checker { return &leaseChecker{leaseStressers} }
+
+func (lc *leaseChecker) Check() error {
+	plog.Info("lease stresser invariant check...")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	errc := make(chan error)
+	for _, ls := range lc.leaseStressers {
+		go func(s Stresser) { errc <- lc.invariantCheck(ctx, s) }(ls)
+	}
+	var errs []error
+	for i := 0; i < len(lc.leaseStressers); i++ {
+		if err := <-errc; err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) == 0 {
+		return nil
+	}
+	return fmt.Errorf("lease stresser encounters error: (%v)", errs[0])
+}
+
+func (lc *leaseChecker) invariantCheck(ctx context.Context, lStresser Stresser) error {
+	ls := lStresser.(*leaseStresser)
+
+	// check revoked leases
+	plog.Debugf("revoked leases %v", ls.revokedLeases)
+	if err := leasesCheck(ctx, false, ls, ls.revokedLeases); err != nil {
+		return err
+	}
+
+	ls.mu.Lock()
+	ls.revokedLeases = make(map[int64]struct{})
+	ls.mu.Unlock()
+
+	if len(ls.aliveLeases) != 0 {
+		// we wait for 2 * TTL to ensure that keepLeaseAlive() does indeed keep live leases alive
+		time.Sleep(TTL * 2 * time.Second)
+	}
+
+	// check alive leases
+	plog.Debugf("alive leases %v", ls.aliveLeases)
+	if err := leasesCheck(ctx, true, ls, ls.aliveLeases); err != nil {
```

Comment:
I don't have preference. Just note that it might not pass one of our linters https://github.com/dominikh/go-simple#checks. But we can change later if that errors.

I am also fine keeping as it is now.